From ce42518db90125765e8b5ecabec24ea7a50bfb7c Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Tue, 9 Apr 2019 19:28:28 +0200
Subject: [PATCH 5/5] recovery: always create emulated volume if we fail to
 detect fs

 * the first fstab entry for data will be used.

Change-Id: I1df70d9df462ab30c857e834152b75d083d77652
---
 volume_manager/VolumeManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/volume_manager/VolumeManager.cpp b/volume_manager/VolumeManager.cpp
index 10a3cda4..3854abaa 100644
--- a/volume_manager/VolumeManager.cpp
+++ b/volume_manager/VolumeManager.cpp
@@ -125,7 +125,7 @@ static int process_config(VolumeManager* vm, fstab_rec** data_recp) {
             if (!*data_recp && !strcmp(fstab->recs[i].mount_point, "/data")) {
                 char* detected_fs_type =
                     blkid_get_tag_value(nullptr, "TYPE", fstab->recs[i].blk_device);
-                if (!strcmp(detected_fs_type, fstab->recs[i].fs_type)) {
+                if (!detected_fs_type || !strcmp(detected_fs_type, fstab->recs[i].fs_type)) {
                     *data_recp = &fstab->recs[i];
                 }
             }
-- 
2.17.1

